File: .tmp_c4000/batch_04_go/3461_check_if_digits_are_equal_in_string_after_operations_i.go
// LeetCode 3461 - Check If Digits Are Equal in String After Operations I
// https://leetcode.com/problems/check-if-digits-are-equal-in-string-after-operations-i/

func hasSameDigits(s string) bool {
	b := []byte(s)
	for len(b) > 2 {
		nb := make([]byte, len(b)-1)
		for i := 0; i+1 < len(b); i++ {
			nb[i] = byte('0' + (int(b[i]-'0')+int(b[i+1]-'0'))%10)
		}
		b = nb
	}
	return b[0] == b[1]
}

File: .tmp_rs4013/rust_check17/3461_check_if_digits_are_equal_in_string_after_operations_i.rs
struct Solution;
// LeetCode 3461 - Check If Digits Are Equal in String After Operations I
// https://leetcode.com/problems/check-if-digits-are-equal-in-string-after-operations-i/

impl Solution {
    pub fn has_same_digits(s: String) -> bool {
        let mut b = s.into_bytes();
        while b.len() > 2 {
            let mut nb = vec![0; b.len() - 1];
            for i in 0..b.len() - 1 {
                nb[i] = b'0' + (b[i] - b'0' + b[i + 1] - b'0') % 10;
            }
            b = nb;
        }
        b[0] == b[1]
    }
}

fn main() {}
